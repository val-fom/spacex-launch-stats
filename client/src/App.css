.App {
  text-align: center;
}

.App-header {
  text-transform: uppercase;  
  padding: 20px 0;
}

.App-content {
  font-size: 8px;
}

.App-launch {
  margin-bottom: 10px;
}
