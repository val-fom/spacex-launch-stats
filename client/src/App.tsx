import React from 'react';
import { ApolloProvider } from '@apollo/react-hooks';
import ApolloClient from 'apollo-boost';
import { BrowserRouter as Router, Switch, Route, Link } from 'react-router-dom';
import { Launches } from './components/Launches';
import './App.css';

const client = new ApolloClient({
  uri: '/graphql',
});

const App: React.FC = () => (
  <Router>
    <ApolloProvider client={client}>
      <div className="App">
        <Link to="/">
          <header className="App-header">SpaceX</header>
        </Link>
        <div className="App-content">
          <Switch>
            <Route path="/" component={Launches} />
          </Switch>
        </div>
      </div>
    </ApolloProvider>
  </Router>
);

export default App;
