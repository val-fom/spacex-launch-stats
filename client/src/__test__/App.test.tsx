import React from 'react';
import { render } from '@testing-library/react';
import App from '../App';

test('renders header', () => {
  const { getByTestId } = render(<App />);
  const header = getByTestId('header');
  expect(header).toBeInTheDocument();
});

test('matches snapshot', () => {
  const { asFragment } = render(<App />);
  expect(asFragment()).toMatchSnapshot();
});
