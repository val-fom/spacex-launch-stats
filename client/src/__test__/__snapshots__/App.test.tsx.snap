// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App matches snapshot in \`Loading\` state 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <header
      class="header"
      data-testid="header"
    >
      <a
        href="/"
      >
        SpaceX
      </a>
    </header>
    <div
      class="content"
    >
      <p
        data-testid="launches-loading"
      >
        Loading...
      </p>
    </div>
  </div>
</DocumentFragment>
`;
